xchange-gdax/src/main/java/org/knowm/xchange/gdax/service/GDAXTradeService.java

```diff
@@ -95,7 +97,21 @@ public TradeHistoryParams createTradeHistoryParams() {
     Collection<Order> orders = new ArrayList<>(orderIds.length);
 
     for (String orderId : orderIds) {
-      orders.add(GDAXAdapters.adaptOrder(super.getOrder(orderId)));
+      try {
+        final GDAXOrder order = super.getOrder(orderId);
+        orders.add(GDAXAdapters.adaptOrder(order));
+      } catch (final ExchangeException e) {
+        // We get 404 for cancelled orders. This is a bit of a pain....
+        if ( e.getCause() instanceof HttpStatusExceptionSupport && ((HttpStatusExceptionSupport)e.getCause()).getHttpStatusCode() == 404) {
+          // We can't actually tell, if we get a 404, whether it was a market or limit order, or
+          // whether it was a bid or an ask and so on. This makes things very tricky.  Let's
+          // put a take in the ground and say you get get neither, and lots of nulls. At least
+          // the caller gets a consistent behaviour they can code to.
+          orders.add(new Order(null, null, null, orderId, null, null, null, Order.OrderStatus.CANCELED) { });
```

Comment:
Faintly horrified by this TBH, but from the caller perspective it actually felt _least_ unpleasant.  Perhaps replace some nulls with zeros, add an `UNKNOWN` `OrderType`... not sure.

Reply:
The ExchangeException should be caught and handled by the client code. That's how every module is implemented.

Reply:
Fair enough. I guess it just means client code supporting GDAX will have to just know that exceptions can be a valid response.